cmd/tetragon/main.go

```diff
@@ -216,13 +217,34 @@ func tetragonExecute() error {
 		obs.RemovePrograms()
 	}()
 
-	go func() {
-		// if we receive a signal, call cancel so that contexts are finalized, which will
-		// leads to normally return from tetragonExecute().
-		s := <-sigs
-		log.Infof("Received signal %s, shutting down...", s)
-		cancel()
-	}()
+	go func(obs *observer.Observer) {
+		finish := false
+		for {
+			// Let's finish without closing sigs channel so we don't panic
+			// on signals on closed channels
+			if finish == true {
+				break
+			}
+
+			s := <-sigs
+			switch s {
+			case syscall.SIGINT, syscall.SIGTERM:
+				// if we receive a signal, call cancel so that contexts are finalized, which will
+				// leads to normally return from tetragonExecute().
+				finish = true
+				log.Infof("Received signal %s, shutting down...", s)
+				cancel()
```

Comment:
Can we just `cancel()` and return here? What is the purpose of setting finish = true, and then breaking out to return?

Reply:
indeed, left over, fixed thank you ;-)